plugins/org.jkiss.dbeaver.ext.hana/src/org/jkiss/dbeaver/ext/hana/model/HANADataSource.java

```diff
@@ -192,7 +198,36 @@ private boolean changeExpiredPassword(DBRProgressMonitor monitor, JDBCExecutionC
             getContainer().getConnectionConfiguration().setUserPassword(passwordInfo.getNewPassword());
             getContainer().persistConfiguration();
         } catch (Exception e) {
-            DBWorkbench.getPlatformUI().showError("Error changing password", "Error changing expired password", e);
+            DBWorkbench.getPlatformUI().showError(HANAMessages.dialog_user_expired_password_error_title, HANAMessages.dialog_user_expired_password_error_message, e);
+            return false;
+        }
+        return true;
+    }
+
+    private boolean changeLicense(DBRProgressMonitor monitor, JDBCExecutionContext context, String purpose) {
+        DBPConnectionConfiguration connectionInfo = getContainer().getActualConnectionConfiguration();
+        String newLicense = DBWorkbench.getPlatformUI().promptText(
+                HANAMessages.dialog_invalid_license_input_title + getContainer().getName(),
+                HANAMessages.dialog_invalid_license_input_label_1 + "\n" + HANAMessages.dialog_invalid_license_input_label_2 + "\n" + 
+                HANAMessages.dialog_invalid_license_input_label_3 + connectionInfo.getUserName() + " - " + connectionInfo.getHostPort() + "\n" + 
+                HANAMessages.dialog_invalid_license_input_label_4, 
```

Comment:
You get the spirit, but let's improve upon that. You can use `org.eclipse.osgi.util.NLS#bind` utility method to bind placeholders to values. You can read more about it here: https://github.com/dbeaver/dbeaver/wiki/Localization#important-notice

Additionally, you can have multiline strings just fine in properties files.